### PR TITLE
Editor: Added samples info for REALISTIC shading

### PR DESCRIPTION
**Pull request by ycw**
This PR added samples info in the viewport info for REALISTIC shading:

https://github.com/mrdoob/three.js/assets/1063018/98e940d9-ecae-453b-90fa-c5d33a0dd338

Preview: https://raw.githack.com/ycw/three.js/editor-viewport-info-samples-info/editor/index.html

### CODE REVIEW

**Comment by ycw** [comment]
just in case you can't find it in the demo video 🙏🏻:

File: editor/js/Viewport.Info.js [hunk 1]
```diff
@@ -98,6 +103,30 @@ function ViewportInfo( editor ) {
...
 
+	//
+
+	editor.signals.pathTracerUpdated.add( function ( samples ) {
+
+		samples = Math.floor( samples );
+
+		samplesText.setValue( samples );
+
+		const samplesStringKey = ( pluralRules.select( samples ) === 'one' ) ? 'viewport/info/oneSample' : 'viewport/info/samples';
```
Comment (Mugen87) [inline editor/js/Viewport.Info.js:114]: Reconsidering #28245.

Do we really need this distinction? Stuff like that creates an unnecessary complexity. Why not just using `Sample(s)`?

Comment (ycw) [reply editor/js/Viewport.Info.js:114]: vertex(ices) then ?

Comment (Mugen87) [reply editor/js/Viewport.Info.js:114]: Or we just always use the plural? I personally don't see `1` samples as an issue which should be fixed with a separate code path.

Comment (ycw) [reply editor/js/Viewport.Info.js:114]: The issue about using `(s)` for just 'Sample(s)' is that, that unit will be inconsistent with other units formatted in the same area:

```
1 object
1 vertex
0 triangles
1 sample(s)
```

The reason that we can't apply `(s)` for other units has been mentioned https://github.com/mrdoob/three.js/pull/28432?notification_referrer_id=NT_kwDOABA4arMxMDcyNTk5ODMxOToxMDYzMDE4#discussion_r1606522848 so skipped.

If the plural rules handling is disturbing, then I prefer taking them into a function:

```
function formats( quantity, unit ) {...}
```

Comment (Mugen87) [reply editor/js/Viewport.Info.js:114]: Then I vote for always using plural (without brackets).

**Comment by ycw** [comment]
https://github.com/mrdoob/three.js/pull/28432/commits/7561b6e581ea0e6deb2f625ca5d41efca43c3c8e looks better now

**Comment by ycw** [comment]
Note that the transition from 0 > 1 > 2 samples is noticeable in low-end devices, so 'always use samples' will confuse users that the quantity is incorrect for `1 samples`

File: editor/js/Viewport.Info.js (hunk 1 above)
Comment (ycw) [reply editor/js/Viewport.Info.js]: Note that the transition from 0 > 1 > 2 samples is noticeable in low-end devices, so 'always use samples' will confuse users that the quantity is incorrect for 1 samples

**Comment by ycw** [comment]
or uses icons with long description as tooltips shown on hover:

image

this way we can eliminate the plural rules totally :D

**Comment by mrdoob** [comment]
Do you mind leaving the plural refactoring for another PR?
This PR is about adding the samples info, but the plural refactoring is blocking it...

**Comment by ycw** [comment]
> Do you mind leaving the plural refactoring for another PR?

ok

File: editor/js/Viewport.Info.js [hunk 2]
```diff
@@ -98,6 +105,30 @@ function ViewportInfo( editor ) {
...
```
Comment (Mugen87) [inline editor/js/Viewport.Info.js:116]: It seems the plural logic is still in place though....

Comment (ycw) [reply editor/js/Viewport.Info.js:116]: > Do you mind leaving the plural refactoring for another PR?

^^^^^^^

:D

**Comment by mrdoob** [comment]
Some clean up: https://github.com/mrdoob/three.js/commit/1cc0bfedfae27073a189f8ac4cabecde5c9f1f68

File: editor/js/Viewport.Info.js (hunk 2 above)
Comment (mrdoob) [reply editor/js/Viewport.Info.js:116]: I think the current plural code is fine (plus the clean up).

The refactoring was over engineering.

**Comment by ycw** [comment]
thanks.

**Pull request merged**